components/tidb_query_executors/src/table_scan_executor.rs

```diff
@@ -408,14 +417,22 @@ impl ScanExecutorImpl for TableScanExecutorImpl {
             table::check_record_key(key)?;
         }
 
-        let some_physical_table_id_column_index = self
+        if let Some(idx) = self
             .column_id_index
-            .get(&table::EXTRA_PHYSICAL_TABLE_ID_COL_ID);
-        if let Some(idx) = some_physical_table_id_column_index {
+            .get(&table::EXTRA_PHYSICAL_TABLE_ID_COL_ID)
+        {
             let table_id = table::decode_table_id(key)?;
             columns[*idx].mut_decoded().push_int(Some(table_id));
             self.is_column_filled[*idx] = true;
         }
+        if let Some(idx) = self.column_id_index.get(&table::EXTRA_COMMIT_TS_COL_ID) {
+            if let Some(ts) = commit_ts {
+                columns[*idx]
+                    .mut_decoded()
+                    .push_int(Some(ts.into_inner() as i64));
+                self.is_column_filled[*idx] = true;
```

Comment:
Could we return an error when `EXTRA_COMMIT_TS_COL_ID` but commit_ts is None? I think it is unexpected